### ISSUE TITLE
BaseModelica generation should probably be idempotent

### ISSUE THREAD

**Issue by bilderbuchi** [OpenModelica/OpenModelica#13393]
We can generate BaseModelica from Modelica by instantiating in OMEdit with `--BaseModelica`. This code in `BM_Idempotent.mo`
```modelica
model BM_Idempotent
Real x;
equation
x = time;
end BM_Idempotent;
```
yields
```modelica
//! base 0.1.0
package 'BM_Idempotent'
  model 'BM_Idempotent'
    Real 'x';
  equation
    'x' = time;
  end 'BM_Idempotent';
end 'BM_Idempotent';
```
Aside from the fact that the model is wrapped in a package, the correspondence is apparent.
Now, creating a new Modelica class and pasting that code in there yields some errors (#13391), but it can be saved in `'BM_Idempotent'.mo` and loaded. 

Now, opening the model `'BM_Idempotent'` in the package `'BM_Idempotent'` (😛), i.e. 
```modelica
model 'BM_Idempotent'
  Real 'x';
equation
  'x' = time;
end 'BM_Idempotent';
``` 
and pressing Instantiate yields the following BaseModelica code:
```modelica
//! base 0.1.0
package ''BM_Idempotent''
  model ''BM_Idempotent''
    Real ''x'';
  equation
    ''x'' = time;
  end ''BM_Idempotent'';
end ''BM_Idempotent'';
```
Here, the apostrophe wrapping is repeated, when it probably should not be -- I got the feeling that converting code that is already BaseModelica into BaseModelica should not change anything, i.e. that operation should be idempotent.

(Also, it is wrapped in a package again, but I assume this is some BaseModelica requirement I am unaware of.)

cc @perost

**Comment by perost** [comment]
This only happens when you instantiate a Base Modelica model using the `--baseModelica` flag, i.e. you get the Base Modelica for a Base Modelica model. The issue is that we currently use the same flag for importing and exporting Base Modelica. In the future we should automatically identify Base Modelica models instead, probably based on the file extension (which I don't think is completely nailed down yet).

That the model is wrapped in a package is indeed a requirement for Base Modelica. The model may contain e.g. records or functions that need to be added to the model. We used to just add those to the model itself, but Base Modelica requires wrapping the model in a package that contains everything instead.

Closing as a duplicate of #12425.
